VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp

```diff
@@ -383,6 +376,34 @@ TArray<TArray<FSplinePoint>> CreateSplinePointsFromInitialFaces(const TArray<FIn
 }
 } // namespace
 
+void FInitialShapeFace::FixOrientation()
+{
+	if (Vertices.Num() < 3)
+	{
+		return;
+	}
+	
+	// Reverse vertices if plane normal points down
+	TArray<FVector3f> VertexPositions;
+	VertexPositions.Reserve(Vertices.Num());
+	for (const auto& Vertex : Vertices)
+	{
+		VertexPositions.Add(FVector3f(Vertex.X, Vertex.Y, Vertex.Z));
+	}
+	FVector3f PlanePointOut;
+	FVector3f PlaneNormalOut;
+	PolygonTriangulation::ComputePolygonPlane(VertexPositions, PlaneNormalOut, PlanePointOut);
+
+	const FVector PlaneNormal(PlaneNormalOut.X, PlaneNormalOut.Y,PlaneNormalOut.Z);
+	float Dot = FVector::DotProduct(FVector::UpVector, PlaneNormal);
+	
+	if (Dot < 0)
+	{
+		Algo::Reverse(Vertices);
+		Algo::Reverse(TextureCoordinates);
+	}
+}

```

Comment:
Nice hidden cleanup ;)

Reply:
Hehe ;) Should maybe have gone into a separate commit!